presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryContext.java

```diff
@@ -60,6 +63,7 @@ public QueryContext(
         this.remoteClientAddress = requireNonNull(remoteClientAddress, "remoteClientAddress is null");
         this.userAgent = requireNonNull(userAgent, "userAgent is null");
         this.clientInfo = requireNonNull(clientInfo, "clientInfo is null");
+        this.clientTags = requireNonNull(clientTags, "clientTags is null");
```

Comment:
ImmutableSet.copyOf?

It's weird we don't do this for session properties a bit below this - we should.

Reply:
Should I change both? Technically it's fine. For session properties, the map was created when calling this. For clientTags, nothing modifies it anyways. But if we want to do `copyOf`, might change it for both?